x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/MachineLearningFlyout.tsx

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiSpacer,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import React, { Component } from 'react';
+import { toastNotifications } from 'ui/notify';
+// @ts-ignore
+import {
+  getMlJob,
+  startMlJob
+} from 'x-pack/plugins/apm/public/services/rest/ml';
+// @ts-ignore
+import { getAPMIndexPattern } from 'x-pack/plugins/apm/public/services/rest/savedObjects';
+import { KibanaLink, ViewMLJob } from 'x-pack/plugins/apm/public/utils/url';
+
+interface FlyoutProps {
+  isOpen: boolean;
+  onClose: () => void;
+  serviceName: string;
+  transactionType?: string;
+  location: any;
+}
+
+interface FlyoutState {
+  isLoading: boolean;
+  hasMlJob: boolean;
+  hasIndexPattern: boolean | null;
+}
+
+export class MachineLearningFlyout extends Component<FlyoutProps, FlyoutState> {
+  public state = {
+    isLoading: false,
+    hasIndexPattern: null,
+    hasMlJob: false
```

Comment:
Now we have `mlAvailable`, `hasMlJob` and `hasDynamicBaseline`. I can't remember if they are all truly different, or if they overlap/are the same.

Reply:
yeah, `mlAvailable` is about the license, `hasMlJob` is about if the current context (service + transaction type) already has a job created. `hasDynamicBaseline` is being replaced with `hasMlJob` :+1: